### DISCUSSION TITLE
Contributing: Formatting and Type Hints

### DISCUSSION THREAD

**Discussion by maflAT** [pyqtgraph/pyqtgraph#2906] [Q&A]
Hello,
I'm using pyqtgraph in a current project and I will make a few changes / additions that I might contribute here.
Before commiting anything I wanted to ask about your stance regarding auto formatting and gradual type hints.

In my project, I have everything auto formatted with black. I see that you also have black configured in your pyproject.toml but most files do not adhere to those settings. Is it ok to reformat an entire module even if i changed just a few lines of code?

I also plan on adding type hints where I might need them. Are you guys ok with inline type annotations, and is it ok if they are incomplete (only part of a module annotated)?

**Comment by j9ac9k** [comment]
Plan to handle formatting will be when I age outstanding PR count is near zero, to minimize the number of merge conflicts.  We would also run it on the whole library at once, and ignore that commit for the purposes of git blame (not sure GitHub supports that feature of git in its blame viewer yet, at one point it did not).

The configuration in pyproject.toml is intended to be used with new files/modules where there won't be any merge conflict resulting in its usage.  If you're adding a new module to the library, by all means make use of that configuration!

Whenever the PR count does get that low, there may be another formatted we want to use (ruff perhaps?) as well, my work on this library has really slowed recently due to other obligations.
